delivery_extra_fee/models/sale_order_line.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    is_delivery_extra = fields.Boolean(related="product_id.is_delivery_extra")
```

Comment:
add readonly=True

Reply:
it is readonly by default, all computed and respectively related fields are readonly

Reply:
I don't think so for related

Reply:
it is, as related is computed by itself, for instance, in the settings all related fields are made explicitly related=False  
https://github.com/odoo/odoo/blob/c236087d126052415e62bd3a725d879d21732c96/addons/account/models/res_config_settings.py#L11